scanner/src/funTest/kotlin/scanners/MultipleScannersTest.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.scanners
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.collections.containExactly
+import io.kotest.matchers.should
+
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageType
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.utils.test.shouldNotBeNull
+
+private val PROJECT_IDENTIFIER = Identifier("Dummy", "", "project", "1.0.0")
+private val PACKAGE_IDENTIFIER = Identifier("Dummy", "", "pkg1", "1.0.0")
+
+class MultipleScannersTest : WordSpec({
+    "Scanning a project and a package with overlapping provenance and non overlapping scanners" should {
+        val analyzerResult = createAnalyzerResult()
+
+        val dummyScanner = DummyScanner()
+        val scannerWrappers = mapOf(
+            PackageType.PROJECT to listOf(dummyScanner.copy(name = "Dummy2")),
+            PackageType.PACKAGE to listOf(dummyScanner)
+        )
+        val ortResult = createScanner(scannerWrappers).scan(analyzerResult, skipExcluded = false, emptyMap())
+
+        "return scan results with non-overlapping scanners" {
+            ortResult.scanner.shouldNotBeNull {
+                val results = getScanResults(PROJECT_IDENTIFIER)
+                results.map { it.scanner.name }.toSet() should containExactly("Dummy2")
```

Comment:
The String `"Dummy2"` is ok, but `"Dummy"` is a bit magic. 
I propose to 1. inline the variable `dummyScanner` and also use the constructor parameter instead of `copy()`. This in turn allows to not make `DummyScanner` a data class.

```
PackageType.PROJECT to listOf(DummyScanner(name = "Dummy1")),
PackageType.PACKAGE to listOf(DummyScanner(name = "Dummy2"))
```